client/src/main/scala/com.crobox.clickhouse/balancing/discovery/ConnectionManagerActor.scala

```diff
@@ -48,8 +52,13 @@ class ConnectionManagerActor(healthProvider: (Uri) => Props, config: Config)
 
     case GetConnection() =>
       if (!initialized) {
-        log.warning("Stashing get connection message until connection message is sent to initialize the manager.")
-        stash()
+        if (fallbackToConfigurationHost) {
+          log.warning("Not yet initialized, returning the config host.")
+          sender ! configHost
+        } else {
+          log.warning("Stashing get connection message until connection message is sent to initialize the manager.")
+          stash()
```

Comment:
Is this favorable? Would it possible cause a burst of useless messages being sent after a cluster takes a while to get up or to respond?

Reply:
We are stashing just until the first connections response is received. 
I do agree that it might not be exactly helpful in some cases but it would not affect the rest of the requests.
We might want to consider the timeout as well and drop the stashed messages after they expire.

Reply:
Yeah id go for that, would make it more durable. Its exactly the logic you would expect from a driver (in terms of graceful degradation in case of trouble)

Reply:
This would be an improvement to the existing behavior and we should tackle it in another pull request.